File: examples/typescript-agent/radius/agent.ts
import { defineAgent } from "@curve-ai/sdk";

const agent = defineAgent({
  name: "typescript-example-agent",
  async run({ text }) {
    return { text: `Radius received: ${text}` };
  },
});

if (process.argv.includes("--radius-dev")) {
  const { serveDevelopmentAgent } = await import("@curve-ai/sdk/development");
  const server = await serveDevelopmentAgent(agent);
  console.log(`Radius ACP development endpoint: ${server.endpoint}`);
} else {
  agent.serveStdio();
}

File: examples/python-agent/radius_python_agent/agent.py
from radius_agent_sdk import RunContext, define_agent, serve_stdio


async def run(context: RunContext) -> str:
    return f"Radius Python received: {context.text}"


agent = define_agent(name="radius-python-example", run=run)

if __name__ == "__main__":
    serve_stdio(agent)
